module/localisation/RobotParticleLocalisation/src/RobotParticleLocalisation.cpp

```diff
@@ -64,48 +73,111 @@ namespace module::localisation {
+                                                           getFieldPosition(goal_post, fd, 0),  // Opp post
+                                                           goals.Hcw);
+                                if (config.debug) {
+
+                                    Eigen::Vector3d state(filter.get());
+                                    Eigen::Transform<double, 3, Eigen::Affine> Hfw;
+                                    Hfw.translation() = Eigen::Matrix<double, 3, 1>(state.x(), state.y(), 0);
+                                    Hfw.linear() =
+                                        Eigen::AngleAxis<double>(state.z(), Eigen::Matrix<double, 3, 1>::UnitZ())
+                                            .toRotationMatrix();
+
+                                    const Eigen::Transform<double, 3, Eigen::Affine> Hcf(goals.Hcw
+                                                                                         * Hfw.inverse().matrix());
+
+                                    const Eigen::Matrix<double, 3, 1> rGCc_own(Hcf
+                                                                               * getFieldPosition(goal_post, fd, 1));
+                                    const Eigen::Matrix<double, 3, 1> rGCc_opp(Hcf
+                                                                               * getFieldPosition(goal_post, fd, 0));
+                                    log(fmt::format("Testing post {}",
+                                                    goal_post.side == Goal::Side::LEFT ? "left" : "Right"));
+                                    log(fmt::format("Candidate own {}", candidate_own));
+                                    log(fmt::format("State {}", state.transpose()));
+                                    log(fmt::format("Hcw {}\n", goals.Hcw));
+                                    log(fmt::format("Actual own post at {}",
+                                                    cartesianToSpherical(rGCc_own).transpose()));
+                                    log(fmt::format("Candidate opp {}", candidate_opp));
+                                    log(fmt::format("Actual opp post at {}",
+                                                    cartesianToSpherical(rGCc_opp).transpose()));
+                                    log(fmt::format("Measured post at {}", m.position.transpose()));
                                 }
+
+                                filter = candidate_own > candidate_opp ? filter_new_own : filter_new_opp;
+                            }
+                            else {
+                                log("Received non-finite measurements from vision. Discarding ...");
                             }
                         }
                     }
+
+                    for (auto goal_post : goals.goals) {
+                        if (goal_post.side == Goal::Side::UNKNOWN_SIDE) {
+                            log<NUClear::INFO>("UNKNOWN_SIDE Goal posts aren't handled yet");
```

Comment:
Action this, or at the absolute least, add a TODO so it's more obvious and easy to find
```suggestion
                            log<NUClear::INFO>("UNKNOWN_SIDE Goal posts aren't handled yet");
```

Reply:
The action to do it is to loop through all 4 possibilities.
Given our vision is able to see pairs of goals fairly often though I'd put this behind a config value if it applies them anyway